rllib/models/tf/fcnet_v2.py

```diff
@@ -4,20 +4,20 @@
 
 import numpy as np
 
-from ray.rllib.models.tf.tf_modelv2 import TFModelV2
+from ray.rllib.models.tf.distributional_q_model import DistributionalQModel
 from ray.rllib.models.tf.misc import normc_initializer, get_activation_fn
 from ray.rllib.utils import try_import_tf
 
 tf = try_import_tf()
 
 
-class FullyConnectedNetwork(TFModelV2):
+class FullyConnectedNetwork(DistributionalQModel):
```

Comment:
This class is a generic model, and should not need to know about DistributionalQModel. The right place to do this is in the catalog.

We already wrap the model for the default models here: https://github.com/ray-project/ray/blob/2a0225dd25a2cb52dec60f9e5e756e3d755bc862/rllib/models/catalog.py#L318

and just need to do it for custom models too: https://github.com/ray-project/ray/blob/2a0225dd25a2cb52dec60f9e5e756e3d755bc862/rllib/models/catalog.py#L260